### PR TITLE
fixed the space preference issue in setting:preference

### LINKED ISSUE #7989: Space setting category doesn't show up
The space setting category doesn't show up. This blocks users from changing settings like "Show all rooms in home space". New app users cannot change this setting at all, forcing them into space filters when interacting with notifications for rooms in spaces (similar to the issue fixed in https://github.com/vector-im/element-android/pull/5827).

Here is where the visibility settings for the spaces category are defined:
https://github.com/vector-im/element-android/blob/6b98b3023e6507b211d849d14dcb167c130cd3d4/vector/src/main/java/im/vector/app/features/settings/VectorSettingsPreferencesFragment.kt#L105-L108

Here is where `isNewAppLayoutFeatureEnabled` is hardcoded to true. This causes `prefs.isVisible` to always be false.
https://github.com/vector-im/element-android/blob/6b98b3023e6507b211d849d14dcb167c130cd3d4/vector/src/main/java/im/vector/app/features/VectorFeatures.kt#L62

### PR DESCRIPTION
**Pull request by criticalAY**
## Type of change

- [ ] Feature
- [x] Bugfix
- [ ] Technical
- [ ] Other :

## Content

Updated the preference file to Fix #7989

## Motivation and context

Fixes #7989

## Screenshots / GIFs
Screenshot_9

## Tests

- Step 1
- Step 2
- Step ...

## Tested devices

- [x] Physical
- [x] Emulator
- OS version(s):

## Checklist

- [x] Changes has been tested on an Android device or Android emulator with API 21
- [x] UI change has been tested on both light and dark themes
- [x] Accessibility has been taken into account. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#accessibility
- [x] Pull request is based on the develop branch
- [x] Pull request includes a new file under ./changelog.d. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#changelog
- [x] Pull request includes screenshots or videos if containing UI changes
- [x] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [ ] You've made a self review of your PR
- [ ] If you have modified the screen flow, or added new screens to the application, you have updated the test [UiAllScreensSanityTest.allScreensTest()](https://github.com/vector-im/element-android/blob/main/vector/src/androidTest/java/im/vector/app/ui/UiAllScreensSanityTest.kt#L73)

Signed-off-by: Ashish Yadav <mailtoashish693@gmail.com>

### CODE REVIEW

**Comment by yostyle** [comment]
@criticalAY thanks for your work. I need to check some points with @mnaturel to approve it.

**Comment by criticalAY** [comment]
Sure!

File: vector/src/main/java/im/vector/app/features/settings/VectorSettingsPreferencesFragment.kt [hunk 1]
```diff
@@ -103,8 +103,8 @@ class VectorSettingsPreferencesFragment :
         }
 
         findPreference<Preference>(VectorPreferences.SETTINGS_PREF_SPACE_CATEGORY)!!.let { pref ->
-            pref.isVisible = !vectorFeatures.isNewAppLayoutFeatureEnabled()
-            pref.isEnabled = !vectorPreferences.isNewAppLayoutEnabled()
+            pref.isVisible = vectorFeatures.isNewAppLayoutFeatureEnabled()
```
Comment (mnaturel) [inline vector/src/main/java/im/vector/app/features/settings/VectorSettingsPreferencesFragment.kt:106]: The setting should not be visible in the new app layout. You can find the fix in #8160

**Comment by mnaturel** [comment]
Hi @criticalAY, thanks for your PR but I am closing it as I have fixed the issue in #8160. You can find [my explanation here](https://github.com/vector-im/element-android/issues/7989#issuecomment-1440219686).

**Pull request closed without merging**